manifests/params.pp

```diff
@@ -195,9 +206,10 @@
   $server_http_allow          = []
 
   # use puppetserver (JVM) or puppet master (Ruby)?
-  $server_implementation = $aio_package ? {
-    true    => 'puppetserver',
-    default => 'master',
+  if $aio_package or ($::osfamily == 'Archlinux') {
+    $server_implementation = 'puppetserver'
```

Comment:
What I'm saying is, that this PR should bring everything needed (not less or more) to configure all the agent related stuff. Master/server support will require in a seperate PR.